e2e-app/src/app/navigation.component.ts

```diff
@@ -1,24 +1,46 @@
 import {Component} from '@angular/core';
+import {Router} from '@angular/router';
 
 import {routes} from './app.routing';
+import {NavigationEnd} from '@angular/router';
 
 @Component({
   selector: 'app-navigation',
   template: `
-  <div ngbDropdown class="d-inline-block">
-    <button class="btn btn-outline-secondary" id="navigation-dropdown" ngbDropdownToggle>Open test suite</button>
-    <div ngbDropdownMenu>
-      <a *ngFor="let url of urls" href="#/{{ url }} " class="dropdown-item" id="navigate-{{ url.replace('/', '-') }}">{{ url }}</a>
+  <a role="button" class="btn btn-outline-primary ml-3" id="navigate-home" href="#/">Menu</a>
+  <div [hidden]="isHidden">
+    <div *ngFor="let group of groupedUrls | keyvalue" class="card m-1 d-inline-block" style="width: 290px;">
+      <div class="card-header">{{group.key}}</div>
+      <ul class="list-group list-group-flush">
+        <li *ngFor="let url of group.value" class="list-group-item">
+          <a href="#{{group.key}}/{{url}}" id="navigate-{{group.key}}-{{url}}" class="list-group-link">{{url}}</a>
+        </li>
+      </ul>
     </div>
   </div>
-  <a role="button" class="btn btn-outline-primary ml-3" id="navigate-home" href="#/">Reset</a>
   `
 })
 
 export class NavigationComponent {
   urls: string[] = [];
+  groupedUrls: any = {};
```

Comment:
To take it a step further, can't we just expose `routes` to the template and simply iterate over them without creating `groupedUrls` ?